fabric-gametest-api-v1/src/main/java/net/fabricmc/fabric/mixin/gametest/StructureTemplateManagerMixin.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.mixin.gametest;
+
+import java.io.IOException;
+import java.util.Optional;
+import java.util.stream.Stream;
+
+import com.google.common.collect.ImmutableList;
+import com.mojang.brigadier.exceptions.CommandSyntaxException;
+import com.mojang.datafixers.DataFixer;
+import org.apache.commons.io.IOUtils;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Shadow;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
+import org.spongepowered.asm.mixin.injection.callback.LocalCapture;
+
+import net.minecraft.block.Block;
+import net.minecraft.nbt.NbtCompound;
+import net.minecraft.nbt.NbtHelper;
+import net.minecraft.registry.RegistryEntryLookup;
+import net.minecraft.resource.Resource;
+import net.minecraft.resource.ResourceFinder;
+import net.minecraft.resource.ResourceManager;
+import net.minecraft.structure.StructureTemplate;
+import net.minecraft.structure.StructureTemplateManager;
+import net.minecraft.util.Identifier;
+import net.minecraft.world.level.storage.LevelStorage;
+
+import net.fabricmc.fabric.impl.gametest.FabricGameTestHelper;
+
+@Mixin(StructureTemplateManager.class)
+public abstract class StructureTemplateManagerMixin {
+	@Shadow
+	private ResourceManager resourceManager;
+
+	@Shadow
+	public abstract StructureTemplate createTemplate(NbtCompound nbt);
+
+	private Optional<StructureTemplate> fabric_loadSnbtFromResource(Identifier id) {
+		Identifier path = FabricGameTestHelper.GAMETEST_STRUCTURE_FINDER.toResourcePath(id);
+		Optional<Resource> resource = this.resourceManager.getResource(path);
+
+		if (resource.isPresent()) {
+			try {
+				String snbt = IOUtils.toString(resource.get().getReader());
+				NbtCompound nbt = NbtHelper.fromNbtProviderString(snbt);
+				return Optional.of(this.createTemplate(nbt));
+			} catch (IOException | CommandSyntaxException e) {
+				throw new RuntimeException("Failed to load GameTest structure %s".formatted(id), e);
```

Comment:
```suggestion
				throw new RuntimeException("Failed to load GameTest structure " + id, e);
```
More concise since the format variable was right at the end